src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,42 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    this.hidden = false;
+    Animal.animalBorn(this);
+  }
+
+  static alive = [];
+
+  static animalBorn(animal) {
+    this.alive.push(animal);
+  }
+
+  static animalDie(animal) {
+    Animal.alive = Animal.alive.filter(element => element !== animal);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(animal) {
+    const canBite = (animal instanceof Herbivore) && animal.hidden === false;
+
+    animal.health = canBite
+      ? animal.health - 50
+      : animal.health;
```

Comment:
do you need this action? I guess you should perform actions only if `canBite`, no other actions required if `!canBite`